### PR TITLE
Lps 90642 experiences

### PR DESCRIPTION
**Pull request by andresfulla**
@sandrodw3

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/experiences.es.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+							nameCurrentValue,
+							priority,
+							segmentsEntryId,
+							segmentsExperienceId
+						} = obj;
+
+						nextState = Object.assign(
+							{},
+							nextState,
+							{
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/experiences.es.js]: use setIn

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/experiences.es.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+ * @param {!string} actionType
+ * @param {object} [payload]
+ * @return {object}
+ * @review
+ */
+export function startCreateExperience(state, actionType, payload) {
+	let nextState = state;
+
+	if (actionType === START_CREATE_EXPERIENCE) {
+		nextState = Object.assign(
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/experiences.es.js]: setIn

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/reducers.es.js [hunk 3]
```diff
@@ -27,15 +30,17 @@ const reducers = [
...
+	selectMappeableTypeReducer,
 	toggleFragmentsEditorSidebarReducer,
 	translationStatusReducer,
 	updateActiveItemReducer,
 	updateDropTargetReducer,
 	updateEditableValueReducer,
 	updateHoveredItemReducer,
-	updateSectionConfigReducer
+	updateSectionConfigReducer,
+	selectExperienceReducer,
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/reducers/reducers.es.js]: ABC order

File: modules/apps/layout/layout-content-page-editor-web/test/js/reducers/experiences.es.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
+import { createExperienceReducer, startCreateExperience, endCreateExperience } from '../../../src/main/resources/META-INF/resources/js/reducers/experiences.es';
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/test/js/reducers/experiences.es.js]: formatter

File: modules/apps/layout/layout-content-page-editor-web/test/js/reducers/experiences.es.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+						segmentsEntryId,
+						segmentsExperienceId: (experiencesCount++, EXPERIENCES_LIST[experiencesCount])
+					}
+				);
+			}
+		};
+	});
+
+	afterAll(() => {
+		global.Liferay = undefiined;
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/test/js/reducers/experiences.es.js]: typo!

Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/test/js/reducers/experiences.es.js]: Liferay is already mocked. Talk to people

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/css/ExperienceSelector.scss [hunk 6]
```diff
@@ -0,0 +1,67 @@
+%ellipsed {
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/css/ExperienceSelector.scss]: `.text-truncate` y `truncate-text`

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+ */
+class ExperienceSelector extends Component {
+
+	/**
+	 * @private
+	 * @review
+	 * @param {Event} event
+	 * @memberof ExperienceSelector
+	 */
+	_createExperience(event) {
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js]: decoupled methods in `_handle` and `_actualFunctionalityFunc`

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.soy [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+							<button
+								class="btn btn-secondary"
+								data-dismiss="modal"
+								data-onclick="{$_closeModal}"
+								type="button"
+							>
+								{msg desc=""}cancel{/msg}
+							</button>
+							<button
+								class="btn btn-primary ml-3"
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.soy]: relate to the modal

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+		this._toggleDropdown();
+	}
+
+	/**
+	 * @private
+	 * @review
+	 * @memberof ExperienceSelector
+	 */
+	_handleDropdownBlur() {
+		clearTimeout(this.willToggleDropdownId);
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js]: requestAnimationFrame

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.soy [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+		<div
+				class="experience-selector position-relative"
+			tabindex="0"
+		>
+			<label for="{$portletNamespace}ExperienceSelector" class="text-white mr-2">
+				{msg desc=""}experience{/msg}
+			</label>
+			<button
+				class="btn btn-sm btn-outline-light mr-2 text-left"
+				onClick="_handleDropdownButtonClick"
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.soy]: data-eventos

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.soy [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+				onBlur="_handleDropdownBlur"
+				type="button"
+				id="{$portletNamespace}ExperienceSelector"
+			>
+				{$activeExperienceLabel}
+			</button>
+			<div
+				class="p-4 rounded dropdown-menu {if $openDropdown}toggled{/if}"
+				onFocus="_handleDropdownFocus"
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.soy]: review

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+		this.willToggleDropdownId = setTimeout(
+			() => {
+				this._toggleDropdown(false);
+			},
+			200
+		);
+	}
+
+	_handleExperienceKeyPress(event) {
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js]: use button inside `li` so you don't need to impllement this

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js [hunk 13]
```diff
@@ -0,0 +1,240 @@
...
+	}
+
+	/**
+	 * @private
+	 * @review
+	 * @param {Event} event
+	 * @memberof ExperienceSelector
+	 */
+	_handleExperienceClick(event) {
+		event.preventDefault();
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js]: don't use preventDefault if you don't need it

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js [hunk 14]
```diff
@@ -0,0 +1,240 @@
...
+		const experienceId = event.delegateTarget.dataset.experienceId;
+		this._selectExperience(experienceId);
+	}
+
+	_showModal(event) {
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js]: reuse show and close. and probably rename

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js [hunk 15]
```diff
@@ -0,0 +1,240 @@
...
+	}
+
+	/**
+	 * Transforms `availableSegments` and `availableExperiences` objects into arrays
+	 * Adds `activeExperienceLabel` to the component state
+	 *
+	 * @inheritDoc
+	 * @review
+	 */
+	prepareStateForRender(state) {
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js]: first inclass methods and the ABC

**Comment by epgarcia** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/fragment_entry_link/FragmentEntryLinkContent.es.js [hunk 16]
```diff
@@ -136,12 +136,12 @@ class FragmentEntryLinkContent extends Component {
 	}
 
 	/**
-	 * Callback executed when languageId property has changed
+	 * Callback executed when experienceId property has changed
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/fragment_entry_link/FragmentEntryLinkContent.es.js]: Quitar este método, y el de `syncLanguageId`

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js [hunk 17]
```diff
@@ -117,7 +117,10 @@ class ExperienceSelector extends Component {
 			{
 				experienceId
 			}
-		);
```
Comment (andresfulla) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/ExperienceSelector.es.js]: en lugar de dispatch, cambia el orden de los reducer e inclute `SELECT_EXPERIENCE` en el reducer de translation status.

**Comment by p2kmgcl** [comment]
Just started reviewing :)

**Comment by andresfulla** [comment]
- [ ] Handlers en Soy as $_...
- [ ] Order CSS class
- [ ] Make sure mehtods are ordered
- [ ] Refactor toggle

**Comment by p2kmgcl** [comment]
Pull request submitted to https://github.com/andresfulla/liferay-portal/pull/43. [See changes here.](https://github.com/p2kmgcl/liferay-portal/compare/168b42e1cbbc~5...168b42e1cbbc)

**Pull request closed without merging**